### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by thgh**
These are just some basic tests, but they are automated and provide a kickstart to build upon.

The test script is using `--single-thread` because there is a race condition between port-authority and livereload.

### CODE REVIEW

File: test/config/entry.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+window.onload = () =>
+  (document.body.innerHTML +=
+    '<br>Path: ' + window.location.pathname + '<br>Dadte: ' + Date.now())
```
Comment (jonkoops) [inline test/config/entry.js:3]: Small typo here.
```suggestion
    '<br>Path: ' + window.location.pathname + '<br>Date: ' + Date.now())
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  deploy:
+    name: Deploy main
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: yarn install
```
[the remark this answers is not in the document]
Comment (thgh) [reply .github/workflows/pr.yml:15]: It does, see https://github.com/thgh/rollup-plugin-livereload/actions/runs/5790693158/job/15694230413

File: package.json [hunk 3]
```diff
@@ -52,11 +52,14 @@
     "livereload": "^0.9.1"
   },
   "devDependencies": {
+    "@rollup/plugin-node-resolve": "^9.0.0",
     "@types/livereload": "^0.9.2",
+    "@types/ws": "^8.5.5",
     "port-authority": "^1.1.1",
-    "rollup": "2",
+    "rollup": "3",
```
[the remark this answers is not in the document]
Comment (thgh) [reply package.json:59]: It's a dev dependency, so this is not installed on the users machine. The code does not depend on rollup so it's not a peerDependency either

File: package.json [hunk 4]
```diff
@@ -52,11 +52,14 @@
     "livereload": "^0.9.1"
   },
   "devDependencies": {
+    "@rollup/plugin-node-resolve": "^9.0.0",
```
[the remark this answers is not in the document]
Comment (thgh) [reply package.json:55]: Thanks, will fix!

File: .github/workflows/pr.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (thgh) [reply .github/workflows/pr.yml:15]: Advantage of leaving it out is that the action runs using the same dependencies as the ones users will have, so if a breaking update to livereload comes out, we will know.

Will add  --frozen-lockfile (equivalent flag for yarn v1)

File: package.json [hunk 5]
```diff
@@ -52,11 +52,14 @@
...
     "@types/livereload": "^0.9.2",
+    "@types/ws": "^8.5.5",
     "port-authority": "^1.1.1",
-    "rollup": "2",
+    "rollup": "3",
     "rollup-plugin-serve": "1",
     "tsup": "^7.2.0",
-    "typescript": "^5.1.6"
+    "typescript": "^5.1.6",
+    "vitest": "^0.34.1"
```
[the remark this answers is not in the document]
Comment (thgh) [reply package.json:63]: Feel free to refactor in @rollup/plugins repo, I just used what I know

File: .github/workflows/pr.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      - run: yarn test
```
[the remark this answers is not in the document]
Comment (thgh) [reply .github/workflows/pr.yml:13]: Hmm, it's not working, will remove for now, but it can be added afterwards!

**Review by jonkoops — APPROVED** [review]

**Pull request merged**